ambry-api/src/main/java/com.github.ambry/account/ContainerBuilder.java

```diff
@@ -0,0 +1,194 @@
+   * @param name The name to set.
+   * @return This builder.
+   */
+  public ContainerBuilder setName(String name) {
+    this.name = name;
+    return this;
+  }
+
+  /**
+   * Sets the status of the {@link Container} to build.
+   * @param status The status to set.
+   * @return This builder.
+   */
+  public ContainerBuilder setStatus(ContainerStatus status) {
+    this.status = status;
+    return this;
+  }
+
+  /**
+   * Sets the description of the {@link Container} to build.
+   * @param description The description to set.
+   * @return This builder.
+   */
+  public ContainerBuilder setDescription(String description) {
+    this.description = description;
+    return this;
+  }
+
+  /**
+   * Sets privacy setting of the {@link Account} to build.
+   * @param isPrivate The privacy setting to set.
+   * @return This builder.
+   */
+  public ContainerBuilder setIsPrivate(boolean isPrivate) {
+    this.isPrivate = isPrivate;
+    return this;
+  }
+
+  /**
+   * Sets the id of the parent {@link Account} of the {@link Container} to build.
+   * @param parentAccountId The parent {@link Account} id to set.
+   * @return This builder.
+   */
+  public ContainerBuilder setParentAccountId(short parentAccountId) {
+    this.parentAccountId = parentAccountId;
+    return this;
+  }
+
+  /**
+   * Builds a {@link Container} object. {@code id}, {@code name}, {@code status}, {@code isPrivate}, and
+   * {@code parentAccountId} are required before build.
+   * @return A {@link Container} object.
+   * @throws IllegalStateException If any required fields is not set.
+   */
+  public Container build() throws JSONException {
+    checkRequiredFields(version);
+    switch (version) {
+      case CONTAINER_METADATA_VERSION_1:
+        JSONObject metadata = new JSONObject();
+        metadata.put(CONTAINER_METADATA_VERSION_KEY, CONTAINER_METADATA_VERSION_1);
+        metadata.put(CONTAINER_ID_KEY, id);
+        metadata.put(CONTAINER_NAME_KEY, name);
+        metadata.put(CONTAINER_STATUS_KEY, status);
+        metadata.put(CONTAINER_DESCRIPTION_KEY, description);
+        metadata.put(CONTAINER_IS_PRIVATE_KEY, isPrivate);
+        metadata.put(CONTAINER_PARENT_ACCOUNT_ID_KEY, parentAccountId);
+        return new Container(metadata);
+
+      default:
+        throw new IllegalStateException("Unsupported container metadata version=" + version);
+    }
+  }
+
+  /**
+   * Checks required fields to build a {@link Container}.
+   * @param version The version to build.
+   */
+  private void checkRequiredFields(short version) {
+    switch (version) {
+      case CONTAINER_METADATA_VERSION_1:
+        if (id == null) {
+          throw new IllegalStateException("Cannot build account, id is not set");
+        }
+        if (name == null) {
+          throw new IllegalStateException("Cannot build account, name is not set");
+        }
+        if (status == null) {
+          throw new IllegalStateException("Cannot build account, status is not set");
```

Comment:
I dont think we need this level of detailed logging. You can do the following
if(id == null || name == null || stats == null || ........................................................)
      throw new IllegalArgumentException("Either of name "+name+" or id "+ id +" ......." is null);

if you fix this, do the same in case of AccountId as well